finquant/quants.py

```diff
@@ -74,6 +75,34 @@ def sharpe_ratio(exp_return, volatility, risk_free_rate=0.005):
     return (exp_return - risk_free_rate) / float(volatility)
 
 
+def value_at_risk(investment, mu, sigma, conf_level=0.95) -> float:
+    """Computes and returns the expected value at risk of an investment/assets.
+
+    :Input:
+     :investment: ``float``/``int``, total value of the investment
+     :mu: ``float``/``int`` average/mean return of the investment
+     :sigma: ``float``/``int`` standard deviation of the investment
+     :conf_level: ``float`` (default= ``0.95``), confidence level of the VaR
+
+    :Output:
+     :Value at Risk: ``float``, VaR of the investment
+    """
+    if not isinstance(
+        investment, (int, float, np.int32, np.int64, np.float32, np.float64)
+    ):
+        raise ValueError("investment is expected to be an integer or float.")
+    if not isinstance(mu, (int, float, np.int32, np.int64, np.float32, np.float64)):
+        raise ValueError("mu is expected to be an integer or float")
+    if not isinstance(sigma, (int, float, np.int32, np.int64, np.float32, np.float64)):
+        raise ValueError("sigma is expected to be an integer or float")
+    if not isinstance(conf_level, float):
+        raise ValueError("confidence level is expected to be a float.")
+    if conf_level >= 1 or conf_level <= 0:
```

Comment:
So the confidence level must not be 0 nor 1?! Just double checking

Reply:
Since the percent point function (namely the inverse of the CDF), which is used to compute the VaR (with `norm.ppf `from `scipy.stats`) has argument `1-conf_level`, using `conf_level = 0` or `conf_level = 1` results in `inf` of `-inf` for the percent point function respectively. I don't know if there are further ways of computing VaR, but I am using the equation provided by the instructor of my QF course. From the real-world perspective, using `conf_level = 0` is equivalent to computing the potential loss that a portfolio can have with a 0% confidence level, and the same goes for `conf_level = 1`. Therefore, it seemed safer to restrict the `conf_level` value between 0 and 1, excluding the two boundary values. What do you think?

Reply:
That sounds very reasonable. Thank you for explaining that :)